iDiff/differs/differs.go

```diff
@@ -0,0 +1,20 @@
+package differs
+
+import "errors"
+
+var diffs = map[string]func(string, string) (string, error){
+	"hist": History,
+	"dir":  Package,
+}
+
+func Diff(arg1, arg2, differ string) (string, error) {
+	if f, exists := diffs[differ]; exists {
+		return callDiffer(arg1, arg2, f)
```

Comment:
I think you should just be able to do `return f(arg1, arg2)` and then avoid needed the `callDiffer` helper function.

as an aside, this might get a bit trickier if we want to add any differs for single images; for example, if we wanted to have a module to detail the structure of a single image. still doable, but just something to keep in mind as you design this.